saleor/graphql/checkout/tests/test_checkout.py

```diff
@@ -1636,6 +1636,52 @@ def test_checkout_no_available_shipping_methods_without_lines(api_client, checko
 }
 """
 
+QUERY_GET_ALL_COLLECTION_POINTS_FROM_CHECKOUT = """
+        query AvailableCollectionPoints(
+            $token: UUID!
+            ){
+            checkout(token: $token){
+                availableCollectionPoints{
+                    name
+                    }
+                }
+            }
+    """
+
+
+def test_available_collection_points_for_preorders_variants_in_checkout(
+    api_client, staff_api_client, checkout_with_preorders_only
+):
+
+    expected_collection_points = list(
+        Warehouse.objects.for_country("US").values("name")
+    )
+    response = staff_api_client.post_graphql(
+        QUERY_GET_ALL_COLLECTION_POINTS_FROM_CHECKOUT,
+        variables={"token": checkout_with_preorders_only.token},
+    )
+    decoded_response = response.json()
+    assert (
+        expected_collection_points
+        == decoded_response["data"]["checkout"]["availableCollectionPoints"]
+    )
+
+
+def test_available_collection_points_for_preorders_and_regular_variants_in_checkout(
+    api_client, staff_api_client, checkout_with_preorders_and_regular_variant
+):
+
+    expected_collection_points = [{"name": "Warehouse2"}]
```

Comment:
It looks like a hardcoded value, does it come from a fixture? If so we could use the fixture here as well.